### PR TITLE
Add documentation page for how to install Grafana and GE on K8s

### PR DESCRIPTION
**Pull request by 09jvilla**
This walks users through how to set up both Grafana and Grafana Enterprise in Kubernetes. 

Once we merge this, we will point the GEM documentation to the Grafana Enterprise in Kubernetes setup, since GE is a requirement for working with GEM. At the moment, GEM users could get stuck on the GE setup step since there is no documentation (we just say that having GE installed is a requirement). 

Our GEM installation is primarily targeted at K8s, so we need a GE installation guide for k8s. 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes Grafana Enterprise 1157
https://github.com/grafana/backend-enterprise/issues/1157 

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/installation/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+### Check that it worked
+Run the following: 
+```bash
+kubectl port-forward service/grafana 3000:3000
+```
+Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. Use `admin` for both the username and password values to login.
+
+A successful Grafana Enterprise installation with a valid license should show an opened lock icon when you mouse over the shield icon in the left menu bar. You can also verify by scrolling down to the bottom of the page.
+![lock_ge](../../lock_ge.png "Showing left menu")
```
Comment (09jvilla) [inline docs/sources/installation/kubernetes.md:237]: I'd like to add 2 images in here, but not sure where I'm supposed to put them. This attempt I was doing was not working (I was trying to put them in grafana/docs) and I don't think its likely the right place. I see some images in grafana/public/img but when I tried putting the images in there, I still had the same path problems that the images wouldn't render.

Comment (oddlittlebird) [reply docs/sources/installation/kubernetes.md:237]: @09jvilla, @osg-grafana or I can show you where to put the images. It's kind of weird, and mentioned in the [Grafana docs folder README](https://github.com/grafana/grafana/tree/master/docs#add-images).

**Review by trevorwhitney — COMMENTED** [review]
Looks great Jen, thanks for writing this up!

**Comment by 09jvilla** [comment]
@osg-grafana ping on this - can I get a quick review? 

Will try to fix the images piece in a bit.

File: docs/sources/installation/kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
+---
+title: Deploy on Kubernetes
+weight: 2
+---
+
+## Deploy Grafana on Kubernetes
+
+This guide will walk you through how to install and run Grafana on Kubernetes (K8S). It will use Kubernetes manifests for the setup. For those who prefer Helm, please see the [Grafana Helm Community Charts](https://github.com/grafana/helm-charts).
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:8]:
```suggestion
This page explains how to install and run Grafana on Kubernetes (K8S). It uses Kubernetes manifests for the setup. If you prefer Helm, refer to the [Grafana Helm community charts](https://github.com/grafana/helm-charts). 
```

File: docs/sources/installation/kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
+---
+title: Deploy on Kubernetes
+weight: 2
+---
+
+## Deploy Grafana on Kubernetes
+
+This guide will walk you through how to install and run Grafana on Kubernetes (K8S). It will use Kubernetes manifests for the setup. For those who prefer Helm, please see the [Grafana Helm Community Charts](https://github.com/grafana/helm-charts). 
+
+For those interested in Grafana Enterprise (not Grafana OS), please jump to [Deploy Grafana Enterprise on Kubernetes](#deploy-grafana-enterprise-on-kubernetes)
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:10]:
```suggestion
If you are interested in Grafana Enterprise (not Grafana OS), jump to [Deploy Grafana Enterprise on Kubernetes](#deploy-grafana-enterprise-on-kubernetes) section.
```

File: docs/sources/installation/kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+#### Create Grafana Kubernetes manifest
+Create a file called `grafana.yaml` and paste the following contents in.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:13]:
```suggestion
1. Create the file `grafana.yaml`, then paste the following contents in it. 
```

File: docs/sources/installation/kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+      protocol: TCP
+      targetPort: http-grafana
+  selector:
+    app: grafana
+  sessionAffinity: None
+  type: LoadBalancer
+```
+
+
+### Send manifest to Kubernetes API Server
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:92]:
```suggestion
### Send manifest to Kubernetes API server
```

Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:92]:
```suggestion
1. Send manifest to Kubernetes API server by runing the following command to create the necessary resources in your cluster.
```

File: docs/sources/installation/kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+Run the following command to create the necessary resources in your cluster.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:94]: Remove

File: docs/sources/installation/kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+
+### Send manifest to Kubernetes API Server
+
+Run the following command to create the necessary resources in your cluster. 
+```bash
+kubectl apply -f grafana.yaml
+```
+
+### Check that it worked
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:99]:
```suggestion
1. Check that it worked by running the following:
```

File: docs/sources/installation/kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+Run the following:
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:100]: Remove

File: docs/sources/installation/kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+```bash
+kubectl apply -f grafana.yaml
+```
+
+### Check that it worked
+Run the following: 
+```bash
+kubectl port-forward service/grafana 3000:3000
+```
+Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. Use `admin` for both the username and password values to login.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:104]:
```suggestion
Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. 

1. Use `admin` for both the username and password to login.
```

File: docs/sources/installation/kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+
+## Deploy Grafana Enterprise on Kubernetes
+The process for deploying Grafana Enterprise is almost identical to the process above, except for some extra steps required to add in your license file.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:107]:
```suggestion
The process for deploying Grafana Enterprise is almost identical to the process above, except for some extra steps required to add in your license file. They are described in the following sections.
```

File: docs/sources/installation/kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+### Obtain Grafana Enterprise License
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:109]:
```suggestion
### Obtain Grafana Enterprise license
```

File: docs/sources/installation/kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+```
+ 
+### Send manifest to Kubernetes API Server
+
+Run the following command to create the necessary resources in your cluster. 
+```bash
+kubectl apply -f grafana.yaml
+```
+
+### Check that it worked
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:229]: Make this a numbered step, as shown above in line 99.

File: docs/sources/installation/kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+    - port: 3000
+      protocol: TCP
+      targetPort: http-grafana
+  selector:
+    app: grafana
+  sessionAffinity: None
+  type: LoadBalancer
+```
+ 
+### Send manifest to Kubernetes API Server
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:222]: Make this a numbered step.

File: docs/sources/installation/kubernetes.md [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+### Obtain Grafana Enterprise License
+To run Grafana Enterprise, you need a valid license. To obtain a license, [contact a Grafana Labs representative](https://grafana.com/contact?about=grafana-enterprise). This guide will assume you already have done this and have a `license.jwt` file. Your license should also be associated with a URL, which we will use further down in this guide. 
+
+### Create License Secret
+Create a Kubernetes secret from your license file using the following command:
+```bash
+kubectl create secret generic ge-license --from-file=/path/to/your/license.jwt
+```
+
+### Create Grafana Enterprise Config
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:118]:
```suggestion
### Create Grafana Enterprise configuration
```

File: docs/sources/installation/kubernetes.md [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+Create a Grafana config file by creating a new file called `grafana.ini` and pasting in the contents below. You will have to update the `root_url` field to the url associated with the license you were given.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:119]:
```suggestion
Create a Grafana configuration file with the name `grafana.ini`. Then paste the content below. 
>**Note:** You will have to update the `root_url` field to the url associated with the license you were given. 
```

File: docs/sources/installation/kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+
+```
+
+### Create Configmap for Grafana Enterprise Config
+Create a Kubernetes Configmap from your `grafana.ini` file with the following command:
+```bash
+kubectl create configmap ge-config --from-file=/path/to/your/config.ini
+```
+### Create Grafana Enterprise Kubernetes manifest
+Create a file called `grafana.yaml` and paste the following contents in. This yaml is identical to the yaml for the OS install except for the additional references to the Configmap that contains your Grafana config file and the Secret that contains your license.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:134]:
```suggestion
Create a `grafana.yaml` file, then paste the content below. This YAML is identical to the one for Grafana OS install except for the additional references to the Configmap which has your Grafana configuration file and the Secret that has your license. 
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added copy-edit suggestions as comments.

File: docs/sources/installation/kubernetes.md [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+```
+Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. 
+
+1. Use `admin` for both the username and password to login.
+
+## Deploy Grafana Enterprise on Kubernetes
+The process for deploying Grafana Enterprise is almost identical to the process above, except for some extra steps required to add in your license file.
+
+### Obtain Grafana Enterprise license
+To run Grafana Enterprise, you need a valid license. To obtain a license, [contact a Grafana Labs representative](https://grafana.com/contact?about=grafana-enterprise). This guide will assume you already have done this and have a `license.jwt` file. Your license should also be associated with a URL, which we will use further down in this guide.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:110]:
```suggestion
To run Grafana Enterprise, you need a valid license. [Contact a Grafana Labs representative](https://grafana.com/contact?about=grafana-enterprise) to obtain the license. This topic assumes that you already have done this and have a `license.jwt` file. Your license should also be associated with a URL, which we will use later in the topic. 
```

**Review by achatterjee-grafana — APPROVED** [review]
Added one more change request. Otherwise, LGTM

**Comment by 09jvilla** [comment]
@achatterjee-grafana I accepted most of your suggestions but I think the formatting is still a bit screwed up as far as when I'm using numbers as opposed to just sub-headers. Can you take another look? I couldn't tell if you wanted numbers in some cases or not others and now the format isn't consistent between the "Grafana" version and the "Grafana Enterprise" version.

File: docs/sources/installation/kubernetes.md [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+kubectl apply -f grafana.yaml
+```
+
+1. Check that it worked by running the following:
+```bash
+kubectl port-forward service/grafana 3000:3000
+```
+Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. 
+
+1. Use `admin` for both the username and password to login.
```
Comment (09jvilla) [inline docs/sources/installation/kubernetes.md:104]: is the idea that I had a 1. here and and 1. immediately prior? or should this be a 2.?

File: docs/sources/installation/kubernetes.md [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+    - port: 3000
+      protocol: TCP
+      targetPort: http-grafana
+  selector:
+    app: grafana
+  sessionAffinity: None
+  type: LoadBalancer
+```
+ 
+1. Send manifest to Kubernetes API Server
```
Comment (09jvilla) [inline docs/sources/installation/kubernetes.md:223]: @achatterjee-grafana here I changed this to a 1. based on your suggestion, but above I have this as a heading with 3 #'s.

File: docs/sources/installation/kubernetes.md [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+---
+
+## Deploy Grafana on Kubernetes
+
+This page explains how to install and run Grafana on Kubernetes (K8S). It uses Kubernetes manifests for the setup. If you prefer Helm, refer to the [Grafana Helm community charts](https://github.com/grafana/helm-charts). 
+
+If you are interested in Grafana Enterprise (not Grafana OS), jump to [Deploy Grafana Enterprise on Kubernetes](#deploy-grafana-enterprise-on-kubernetes) section.
+
+#### Create Grafana Kubernetes manifest
+1. Create the file `grafana.yaml`, then paste the following contents in it.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:13]:
```suggestion
1. Create the `grafana.yaml` file, then paste the content below. 
```

File: docs/sources/installation/kubernetes.md [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+      targetPort: http-grafana
+  selector:
+    app: grafana
+  sessionAffinity: None
+  type: LoadBalancer
+```
+
+
+### Send manifest to Kubernetes API server
+
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:93]: 1. Run the following command:

File: docs/sources/installation/kubernetes.md [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+  type: LoadBalancer
+```
+ 
+1. Send manifest to Kubernetes API Server
+
+```bash
+kubectl apply -f grafana.yaml
+```
+
+2. Check that it worked
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:229]:
```suggestion
1. Check that it worked by running the following:
```

File: docs/sources/installation/kubernetes.md [hunk 23]
```diff
@@ -0,0 +1,233 @@
...
+```bash
+kubectl port-forward service/grafana 3000:3000
+```
+Now if you navigate to `localhost:3000` in your browser, you should see a Grafana login page. Use `admin` for both the username and password values to login.
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:233]:
```suggestion
Now if you navigate to `localhost:3000` in your browser, you should see the Grafana login page. 

1. Use `admin` for both the username and password to login.
```

File: docs/sources/installation/kubernetes.md (hunk 18 above)
Comment (achatterjee-grafana) [reply docs/sources/installation/kubernetes.md:104]: All numbered list should start with 1. They get rendered correctly. For more information, check the style guide :)

**Review by achatterjee-grafana — COMMENTED** [review]
Just a few more, and I have fixed the numbering and other inconsistencies.

**Comment by DanCech** [comment]
@09jvilla lots of small commits is very hard on the ci/cd system, please try to group your changes into fewer commits.

**Comment by 09jvilla** [comment]
Sorry @DanCech . The formatting I'm seeing in my browser vs the markdown formatter I was using don't seem to match. I'll pick this up later.

**Comment by 09jvilla** [comment]
@oddlittlebird I think I'm basically good to go on content except I noticed one issue -- I'd like the page I added to be at the bottom of the left hand menu, not the top. I'm not sure how I control that? I'm guessing its something in the website repo? 

Basically I mean that "Requirements" should still be the first subheader under installation. 

Screen Shot 2021-03-25 at 3 48 50 PM

**Comment by achatterjee-grafana** [comment]
@09jvilla use the weight metadata, weight = 700:

File: docs/sources/installation/kubernetes.md [hunk 24]
```diff
@@ -0,0 +1,228 @@
+---
+title: Deploy on Kubernetes
```
Comment (achatterjee-grafana) [inline docs/sources/installation/kubernetes.md:2]: +++
title = "Deploy Grafana on Kubernetes"
description = "Guide for deploying Grafana on Kubernetes"
keywords = ["grafana", "configuration", "documentation", "kubernetes"]
weight = 700
+++

**Review by achatterjee-grafana — COMMENTED** [review]
Added metadata information for you to use.

**Comment by 09jvilla** [comment]
Alrighty I think we're ready to go @achatterjee-grafana . What's the next step? Looks like I'm not authorized to merge.

**Comment by achatterjee-grafana** [comment]
Done! Thanks for adding this topic!!

**Comment by osg-grafana** [comment]
> @osg-grafana ping on this - can I get a quick review?
> 
> Will try to fix the images piece in a bit.

This was at the top of my pile today, as you mentioned a deadline of this week. I see there has been assistance from the Docs Squad already. Glad that this is merged in. Thank you for jumping in @achatterjee-grafana and @oddlittlebird!

**Pull request merged**